### PR TITLE
Modified comparison of Match instances - issue230 and issue246

### PR DESCRIPTION
**Pull request by jllorente**
Using the internal fields of a Match object `match.u.user.name` may lead to issues when using deprecated and aliased modules. For example, state vs conntrack.
This PR reduces the fields used to compare Match objects in favor of the actual name of the iptables rule.
The test of the state match is also updated to reflect this change.

### CODE REVIEW

**Comment by ldx** [comment]
It seems that the python 2.6 environment on Travis CI is currently broken.

Otherwise, this looks good :+1: Any chance to add a test that would have failed with the old logic, but passes now with this PR?

File: tests/test_matches.py [hunk 1]
```diff
@@ -389,8 +389,10 @@ def test_state(self):
         self.chain.insert_rule(self.rule)
         rule = self.chain.rules[0]
         m = rule.matches[0]
-        self.assertEquals(m.name, "state")
-        self.assertEquals(m.state, "RELATED,ESTABLISHED")
+        self.assertEqual(m.name, "state")
+        self.assertEqual(m.state, "RELATED,ESTABLISHED")
+        self.assertEqual(rule.matches[0].name, self.rule.matches[0].name)
+        self.assertEqual(rule, self.rule)
```
Comment (jllorente) [inline tests/test_matches.py]: @ldx , this is the test that would fail with older versions since it compares head to head the 2 rules.

Comment (jllorente) [reply tests/test_matches.py]: Line 395: self.assertEqual(rule, self.rule)

**Comment by ldx** [comment]
Can you please rebase this on master to fix the Travis CI build?

**Pull request merged**